src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeAlleleCounts.java

```diff
@@ -371,13 +371,12 @@ public String toUnphasedGenotypeString() {
             return "";
         }
         final StringBuilder sb = new StringBuilder(distinctAlleleCount * 3);
-        for (int i = 0; i < distinctAlleleCount; i += 2) {
+        for (int i = 0; i < distinctAlleleCount * 2; i += 2) {
             final int alleleIndex = sortedAlleleCounts[i];
             final int alleleCount = sortedAlleleCounts[i + 1];
             for (int j = 0; j < alleleCount; j++) {
```

Comment:
Since counts is guaranteed not to be 0, then the string composition could be  rather:

```
sb.append(index);
for (int j = 1; j < count; j++) {
    sb.append('/').append(index);
}
return sb.toString();
```

Reply:
no, you need to add a `/` after every allele, not only the alleles with counts >1. think about 0/1, whose sortedAlleleCounts array is (0,1,1,1)....

Reply:
index is 0 and 1, count is 1 and 1. ... (index=0, count=1) ... then (index=1, count=1). count is always 1 or larger or that is my understanding.... 

at least in the example that you are giving count for both alleles (0 and 1) is 1 (the second number of the pair. (0, 1, 1, 1) is decomposed into (0, 1) and (1, 1)

Reply:
In any case you could leave the code as it is... just noticed this optimization opportunity near your change so I suggested the edit but it has no concern with fix, so.